### PR TITLE
chore: privatizing public API of integrations in flask/patch.py

### PR DESCRIPTION
**Pull request by rachelyangdog**
The API should just be what is necessary. A lot of integrations expose implementation details through the patch.py We can’t just remove these functions right away as it will break our API so for each we will need to deprecate first and then remove in 2.x.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/contrib/flask/patch.py [hunk 1]
```diff
@@ -70,11 +72,21 @@
 )
 
 
-def get_version():
+def _get_version():
     # type: () -> str
     return get_version_for_package("flask")
 
 
+def get_version():
```
Comment (mabdinur) [inline ddtrace/contrib/flask/patch.py:80]: `get_version` is defined in every integration. If we plan to rename this function in the flask integration we should do it for our other integrations. Here's the PR that introduced this function: https://github.com/DataDog/dd-trace-py/commit/ab6321c9dd2beccecbc8e8028cd4144689db98fe.

Comment (ZStriker19) [reply ddtrace/contrib/flask/patch.py:80]: I think the plan is to go one integration at a time. @rachelyangdog lmk if that sounds right.

Comment (ZStriker19) [inline ddtrace/contrib/flask/patch.py:80]: We need to make sure that everywhere any of the now deprecated methods are called, we switch them to the non-deprecated private version. E.g. we need to change line 111. This'll fix a lot of the failing tests.

Comment (rachelyangdog) [reply ddtrace/contrib/flask/patch.py:80]: yes! i was going to make a PR for each integration

File: ddtrace/contrib/flask/__init__.py [hunk 2]
```diff
@@ -108,6 +108,6 @@ def index():
 
         patch = _patch.patch
         unpatch = _patch.unpatch
-        get_version = _patch.get_version
+        get_version = _patch._get_version
```
Comment (ZStriker19) [inline ddtrace/contrib/flask/__init__.py:111]: We shouldn't change this. The get_version we expose here should raise a deprecation warning.

**Comment by erikayasuda** [comment]
@DataDog/apm-idm-python Could we have a reviewer on this? Sorry the automation didn't automatically tag someone

**Review by ZStriker19 — APPROVED** [review]

File: ddtrace/contrib/flask/patch.py [hunk 3]
```diff
@@ -71,11 +73,21 @@
 )
 
 
-def get_version():
```
Comment (mabdinur) [inline ddtrace/contrib/flask/patch.py:74]: Can we revert this change? get_version is defined every integration and is used in ddtrace._monkey. Renaming this method would require a significant change that will updated tens of files.
